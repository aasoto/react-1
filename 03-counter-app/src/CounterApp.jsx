import { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({value}) => {

  const [ counter, setCounter ] = useState( value );

  const handleAdd = () => {
    setCounter((c) => {
      return c + 1;
    })
  }

  const handleSubtract = () => {
    setCounter((c) => {
      return c - 1;
    })
  }

  const handleReset = () => {
    setCounter(value)
  }

  return (
    <>
      <h1>Counter App</h1>
      <h2 aria-label='counter'> { counter } </h2>

      <button aria-label='increment' onClick={ handleAdd }> +1 </button>
      <button aria-label='decrement' onClick={ handleSubtract }> -1 </button>
      <button aria-label='reset' onClick={ handleReset }> Reset </button>

    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}