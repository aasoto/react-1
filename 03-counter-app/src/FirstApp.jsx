const newMessage = {
  message: 'Bienvenidos a mi primera aplicación.',
  title: ' Andrés Soto React'
}

const getMessage = () => {
  return newMessage.message;
}

const getTitle = () => {
  return newMessage.title;
}

const sumar = (a, b) => {
  return a + b;
}

export const FirstApp = () => {
  /**
   * Para agrupar un numero de elementos debe haber un nodo padre,
   * el cual se puede definir con la etiqueta <Fragment></Fragment>
   * de manera resumida <></>
   */
  return (
    <>
      <h1>{getTitle()}</h1>
      <h3>{sumar(5, 3)}</h3>
    </>
  );
}