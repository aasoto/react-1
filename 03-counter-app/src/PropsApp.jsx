import PropTypes from 'prop-types'

export const PropsApp = ({title, subTitle, developper}) => {
  return (
    <>
      <a href="#">
        <h1 data-testid="test-title">{title}</h1>
      </a>
      <h4>{subTitle}</h4>
      <h4>{subTitle}</h4>
      <p>Desarrollador por: {developper}</p>
    </>
  );
}

PropsApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  developper: PropTypes.string
}

PropsApp.defaultProps = {
  developper: 'Andrés Soto',
  subTitle: 'No hay subtítulo',
}