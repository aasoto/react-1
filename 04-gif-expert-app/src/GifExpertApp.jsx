import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['One Punch', 'Dragon Ball'])

  const onAddCategory = () => {
    setCategories(['Valorant', ...categories]);
  }

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategory setCategories={setCategories}/>
      <button onClick={onAddCategory}>Agregar categor√≠a</button>
      <ol>
        {
          categories.map(category => {
            return <li key={category}>{category}</li>
          })
        }
      </ol>
    </>
  );
}