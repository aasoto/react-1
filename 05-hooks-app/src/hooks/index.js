export * from './useCounter';
export * from './useFetch';
export * from './UseForm';
export * from './useTodo';
