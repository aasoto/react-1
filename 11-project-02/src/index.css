@tailwind base;
@tailwind components;
@tailwind utilities;

.navbar {
  @apply fixed z-20 flex justify-center items-center w-full h-24 p-4
}

.navbar nav{
  @apply w-full h-full backdrop-blur-md bg-white/50 rounded-lg px-5 md:px-10 py-2 shadow-lg
}

.navbar .nav-container{
  @apply flex justify-between items-center
}

.navbar .nav-section{
  @apply flex items-center gap-5
}

.navbar h1{
  @apply text-black font-bold text-2xl italic
}