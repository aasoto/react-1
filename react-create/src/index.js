// Variables y constantes

const nombre = 'AndrÃ©s';
let