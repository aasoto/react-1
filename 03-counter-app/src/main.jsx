import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import HelloWorldApp from './HelloWorld';
import { PropsApp } from './PropsApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <PropsApp title="Probando PropTypes" subTitle='Esto es un subtÃ­tulo'/>
    {/* <CounterApp value={4} /> */}
  </React.StrictMode>
);