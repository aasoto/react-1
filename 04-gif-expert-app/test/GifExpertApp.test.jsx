import { fireEvent, getByText, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Pruebas del <GifExpertApp/>', () => {

  test('debe hacer match con el snapshot', () => {
    
    const {container} = render( <GifExpertApp/>);
    expect(container).toMatchSnapshot();

  });

  test('debe aceptar categoría', () => {
    render(<GifExpertApp/>);

    const input = screen.getByRole('textbox');
    const form  = screen.getByRole('form', {name: 'main-form'});

    fireEvent.input(input, { target: { value: 'Saitama' } });
    fireEvent.submit(form);

    expect(input.value).toBe('');
    expect(screen.getByText('Saitama'));

  });

  test('debe rechazar categoría repetida', () => {
    render(<GifExpertApp/>);

    const input = screen.getByRole('textbox');
    const form  = screen.getByRole('form', {name: 'main-form'});

    fireEvent.input(input, { target: { value: 'Saitama' } });
    fireEvent.submit(form);

    expect(input.value).toBe('');
    expect(screen.getByText('Saitama'));

    fireEvent.input(input, { target: { value: 'Saitama' } });
    fireEvent.submit(form);

    expect(input.value).toBe('');
    expect(screen.getAllByText('Saitama').length).toBe(1);
    // screen.debug();
  });

});