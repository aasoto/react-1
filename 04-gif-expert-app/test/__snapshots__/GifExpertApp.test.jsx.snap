// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del <GifExpertApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="main-form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Dragon Ball
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
